src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -50,7 +51,7 @@ public synchronized void clearCurrentCommand() {
     }
 
     public synchronized void runCurrentCommand() {
-        currentCommand.execute(driverManager.getDriverToDraw());
+        currentCommand.execute(driverManager.getExtensionDriver().getCurrentDriver());
```

Comment:
You should not change the API. Originally there was getCurrentDriver() and it should be enough:

- getCurrentDriver should return driver composite and it should simply be able to delegate all requests to child (horizontal or extension) sub-drivers.
- setCurrentDriver should delegate a request to set driver to driver composite so it could manage it. 
- you might need to replece some getCurrentDriver and setCurrentDriver when they are used e.g. to apply decorator (because driver composite with your extensions should be able to manage them)